exchanges/bybit/bybit_websocket.go

```diff
@@ -107,443 +113,769 @@ func (by *Bybit) WsAuth(ctx context.Context) error {
+			Price:        result[x].Price.Float64(),
+			Amount:       result[x].Size.Float64(),
+			Side:         side,
+			TID:          result[x].TradeID,
+		}
+	}
+	return trade.AddTradesToBuffer(by.Name, tradeDatas...)
 }
 
-func (by *Bybit) wsUpdateOrderbook(update *WsOrderbookData, p currency.Pair, assetType asset.Item) error {
-	if update == nil || (len(update.Asks) == 0 && len(update.Bids) == 0) {
-		return errors.New("no orderbook data")
+func (by *Bybit) wsProcessOrderbook(assetType asset.Item, resp *WebsocketResponse) error {
+	var result WsOrderbookDetail
+	err := json.Unmarshal(resp.Data, &result)
+	if err != nil {
+		return err
 	}
-	asks := make([]orderbook.Item, len(update.Asks))
-	for i := range update.Asks {
-		target, err := strconv.ParseFloat(update.Asks[i][0], 64)
+	cp, err := currency.NewPairFromString(result.Symbol)
+	if err != nil {
+		return err
+	}
+	asks := make([]orderbook.Item, len(result.Asks))
+	for i := range result.Asks {
+		var price float64
+		price, err = strconv.ParseFloat(result.Asks[i][0], 64)
 		if err != nil {
 			return err
 		}
-		amount, err := strconv.ParseFloat(update.Asks[i][1], 64)
+		var size float64
+		size, err = strconv.ParseFloat(result.Asks[i][1], 64)
 		if err != nil {
 			return err
 		}
-		asks[i] = orderbook.Item{Price: target, Amount: amount}
+		asks[i] = orderbook.Item{Price: price, Amount: size}
 	}
-	bids := make([]orderbook.Item, len(update.Bids))
-	for i := range update.Bids {
-		target, err := strconv.ParseFloat(update.Bids[i][0], 64)
+	bids := make([]orderbook.Item, len(result.Bids))
+	for i := range result.Bids {
+		var price float64
+		price, err = strconv.ParseFloat(result.Bids[i][0], 64)
 		if err != nil {
 			return err
 		}
-		amount, err := strconv.ParseFloat(update.Bids[i][1], 64)
+		var size float64
+		size, err = strconv.ParseFloat(result.Bids[i][1], 64)
 		if err != nil {
 			return err
 		}
-		bids[i] = orderbook.Item{Price: target, Amount: amount}
+		bids[i] = orderbook.Item{Price: price, Amount: size}
 	}
-	return by.Websocket.Orderbook.LoadSnapshot(&orderbook.Base{
-		Bids:            bids,
-		Asks:            asks,
-		Pair:            p,
-		LastUpdated:     update.Time.Time(),
-		Asset:           assetType,
-		Exchange:        by.Name,
-		VerifyOrderbook: by.CanVerifyOrderbook,
-	})
+	if len(asks) == 0 && len(bids) == 0 {
+		return nil
+	}
+	if resp.Type == "snapshot" {
+		err = by.Websocket.Orderbook.LoadSnapshot(&orderbook.Base{
+			Pair:         cp,
+			Exchange:     by.Name,
+			Asset:        assetType,
+			LastUpdated:  result.UpdateTime.Time(),
+			LastUpdateID: result.Sequence,
+			Asks:         asks,
+			Bids:         bids,
+		})
+		if err != nil {
+			return err
+		}
+	} else {
+		err = by.Websocket.Orderbook.Update(&orderbook.Update{
+			Pair:       cp,
+			Asks:       asks,
+			Bids:       bids,
+			Asset:      assetType,
+			UpdateID:   result.Sequence,
+			UpdateTime: result.UpdateTime.Time(),
+		})
+		if err != nil {
+			return err
+		}
```

Comment:
Please get a test that handles orderbook update too

Reply:
https://github.com/thrasher-corp/gocryptotrader/blob/2933cd64feecd97ffb05fb6d739b6b0eb45f2597/exchanges/bybit/bybit_test.go#L2967

There is here.

Reply:
But there isn't a order book delta update 👀

Reply:
Orderbooks respond with `delta` updates that are not full snapshots
`{"topic":"orderbook.50.BTCUSDT","ts":1697509522507,"type":"delta","data":{"s":"BTCUSDT","b":[["28300.86","2.814922"]],"a":[],"u":1424925,"seq":15164802540}}`
Looks like your implementation is correct, but please add test coverage